File: src/SkyBlock/enchants/sword/Disarm.php
<?php


namespace SkyBlock\enchants\sword;


use pocketmine\event\entity\EntityDamageEvent;
use pocketmine\item\VanillaItems;
use pocketmine\player\Player;
use SkyBlock\enchants\BaseEnchantment;

class Disarm extends BaseMeleeEnchant {

    static int $id = 114;

    /**
     * @param Player $holder
     *
     * @return bool
     */
    public function isApplicableTo(Player $holder) : bool {
        return mt_rand(1, 50) === 1;
    }

    public function onActivation(Player $player, Player $attacker, EntityDamageEvent $ev, int $enchantmentlevel) : void {
        $hitem = $attacker->getInventory()->getItemInHand();
        if (!BaseEnchantment::hasAnyEnchantment($hitem, [DisarmProtection::$id, Serpent::$id, Brawler::$id, Wizardly::$id, Disorder::$id, Thunderbolt::$id, Disarmor::$id, Detonate::$id, Potshot::$id, Smasher::$id, SoulSnatcher::$id])) {
            $attacker->getInventory()->setItemInHand(VanillaItems::AIR());
            $attacker->getWorld()->dropItem($attacker->getPosition()->asVector3(), $hitem);
            $this->sendActivation($player, "§bDisarm §aActivated!");
            $this->sendActivation($attacker, "§cYOU HAVE BEEN DISARMED!");
        } else {
            $this->sendActivation($player, "§bDisarm §cDeactivated! Player has Disarm Protection or an exclusive enchant!");
        }
    }

}

File: SkyblockCore/src/main/java/dev/skyblock/enchants/sword/Disarm.java
package dev.skyblock.enchants.sword;

import dev.skyblock.enchants.BaseMeleeEnchant;
import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;
import org.bukkit.potion.PotionEffect;
import org.bukkit.potion.PotionEffectType;
import java.util.concurrent.ThreadLocalRandom;

public class Disarm extends BaseMeleeEnchant {
    public static final int ID = 114;

    public Disarm() {
        super(ID, "Disarm", 5);
    }

    @Override
    public boolean isApplicableTo(Player holder, int level) {
        return ThreadLocalRandom.current().nextInt(1, 26) == 1;
    }

    @Override
    public void onActivation(Player victim, Player attacker, EntityDamageByEntityEvent ev, int level) {

    }
}
